sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -192,6 +192,28 @@ class DataFrameSuite extends QueryTest
       structDf.select(xxhash64($"a", $"record.*")))
   }
 
+  private def assertDecimalSumOverflow(
+      df: DataFrame, ansiEnabled: Boolean, expectedAnswer: Row): Unit = {
+    if (!ansiEnabled) {
+      try {
+        checkAnswer(df, expectedAnswer)
+      } catch {
```

Comment:
- This has changed the tests to say it is ok to throw exception even when ansienabled is false.   Our ansienabled flag then isn't doing what it says it is supposed to do, right.  That is a bug.

Reply:
This is a bug, in unsafe row writer, and I think we should fix it. According to https://github.com/apache/spark/pull/27627#issuecomment-591185451 , this bug is already there for a long time.

This is a less critical bug as Spark fails instead of returning a wrong result.

Reply:
Just to clarify, the issue we have is the UnsafeRow taking a overflow decimal value but when fetching it throws exception.